### ISSUE TITLE
Better support micro-frontend

### ISSUE THREAD

**Issue by oliviertassinari** [mui/mui-toolpad#1183]
### Duplicates

- [X] I have searched the existing issues

### Latest version

- [X] I have tested the latest version

### Summary 💡

Per the https://micro-frontends.org/ architecture, I would like to be able able to build some parts of my app in a low-code fashion with Toolpad, and others with React/pro-code. However, the current layout is optimized for a full scale application.

### Examples 🌈

Here is how it could look like without the padding left and right, the margin-top, and the `overflow: scroll`.

### Before

<img width="535" alt="Frame 2 (1)" src="https://user-images.githubusercontent.com/3165635/196937974-7cb72d0c-b604-4827-8fe7-d20336d9174a.png">

### After

<img width="558" alt="Frame 1 (1)" src="https://user-images.githubusercontent.com/3165635/196937986-a6a23b4f-528c-4eca-9474-7ece3476b7f2.png">

### Motivation 🔦

Save space

**Comment by Janpot** [comment]
We wrap the whole page in a MUI Container. Is this a matter of adding an option on the page to disable that Container?

**Comment by oliviertassinari** [comment]
@Janpot yeah, it could be enough. Another option would be to expose a React component that I can render inside the Zendesk extension, on which I can then apply CSS overrides 😁 or not come without these default styles, or have a prop to disable the style.

**Comment by oliviertassinari** [comment]
@Janpot More options:

3. Expose a runtime component and sync with API calls. This is the approach used Builder.io, e.g. https://www.builder.io/c/docs/integrate-section-building, or Plasmic https://docs.plasmic.app/learn/nextjs-quickstart/, MakeShift https://www.makeswift.com/docs/guides/manual-setup. The problem with this approach is that unless we can lazily load components, this will be heavy on the page.

```jsx
import { BuilderComponent, builder } from '@builder.io/react';

// Replace with your Public API Key.
builder.init(YOUR_API_KEY);

export async function getStaticProps({ params }) {
  const urlPath = '/' + (params?.page?.join('/') || '');
  const announce = await builder
    .get('announcement-bar', { userAttributes: { urlPath } })
    .toPromise();

  return {
    props: {
      announce: announce || null,
    },
  };
}

export default function Page({ announce }) {
  return (
    <>
      {/* Put your header here. */}
      <YourHeader />
      <BuilderComponent model="announcement-bar" content={announce} />
      {/* Put the rest of your page here. */}
      <TheRestOfYourPage />
    </>
  );
}
```

5. Have a `_generated` folder developers can import from. This is the approach used by https://www.convex.dev/. The code would be committed to git. Any changes done outside of Toolpad are erased. 

```jsx
src/MyApp.jsx
import { useQuery } from "../convex/_generated/react";
import { useMutation } from "../convex/_generated/react";

function MyApp() {
  // data will be `undefined` while the query is first loading
  const data = useQuery("listMessages");
  const sendMessage = useMutation("sendMessage");
  const sendHello = () => sendMessage("Hello!", "me");
  return (
    <div>
      <MyListView data={data} />
      <button onClick={sendHello}>click me!</button>
    </div>
  );
}
```

**Comment by Janpot** [comment]
I propose we improve the story for iframe embedding as part of MVP.

- Support disabling the container component through page level parameter
- Support disabling the navigation through query params** as part of https://github.com/mui/mui-toolpad/issues/813 
- Anything else?

Then leave exporting components for later. Personally, 4. makes most sense to me. 


** Let's call this the Toolpad chrome, or Toolpad shell? It also includes header bar for instance.

**Comment by oliviertassinari** [comment]
> Toolpad shell

This makes more sense to me, as a reference to app shell: https://developer.chrome.com/blog/app-shell/.

> Query parameter

This would be more universal, it's not always possible or easy to add an iframe attribute. I'm not even sure that iframe attributes would fly, since iframe will likely be always for when the origin is different https://stackoverflow.com/questions/13570733/html5-data-attributes-in-iframe-element

> Personally, 4. makes the most sense to me.

My only concern is that option 4. doesn't lazy load, which will be a scaling nightmare. If it's simpler to pull off than option 1 then it sounds like a great tradeoff, as an MVP.

**Comment by Janpot** [comment]
👍  "Toolpad shell" it is then.

Right, data attributes won't work then. My only concern with using query parameters to pass toolpad options to the application is that they may collide with user defined parameters. But maybe it's just a matter of namespacing them well. (`?toolpad.display=standalone`)

> My only concern is that option 4. can't support lazy loading

What do you mean by lazy loading? Wouldn't it be possible to wrap whatever you want to lazy load in a component and import it with `React.lazy`?

**Comment by oliviertassinari** [comment]
> What do you mean by lazy loading?

I have changed the terminology, it was confusing ~loading~ bundling.
